bin/light-base/src/runtime_service.rs

```diff
@@ -1489,25 +1486,37 @@ impl<TPlat: Platform> Background<TPlat> {
                         let all_blocks_notif = Notification::Finalized {
                             best_block_hash,
                             hash: finalized_block.hash,
-                            pruned_blocks: pruned_blocks
-                                .into_iter()
-                                .map(|(_, b, _)| b.hash)
-                                .collect(),
+                            pruned_blocks: pruned_blocks.iter().map(|(_, b, _)| b.hash).collect(),
                         };
 
                         let mut to_remove = Vec::new();
                         for (subscription_id, (sender, finalized_pinned_remaining)) in
                             all_blocks_subscriptions.iter_mut()
                         {
-                            if *finalized_pinned_remaining == 0 {
+                            let count_limit = pruned_blocks.len() + 1;
+
+                            if *finalized_pinned_remaining < count_limit {
                                 to_remove.push(*subscription_id);
                                 continue;
                             }
 
-                            if sender.try_send(all_blocks_notif.clone()).is_ok() {
-                                *finalized_pinned_remaining -= 1;
-                            } else {
+                            if sender.try_send(all_blocks_notif.clone()).is_err() {
                                 to_remove.push(*subscription_id);
+                                continue;
+                            }
+
+                            *finalized_pinned_remaining -= count_limit;
+
+                            // Mark the finalized and pruned blocks as finalized or non-canonical.
```

Comment:
```suggestion
                            // Mark the finalized and pruned blocks as finalized and non-canonical.
```

?

Reply:
By "non-canonical", I mean "not part of the finalized chain". So by definition "finalized" means "canonical".